yjit_codegen.c

```diff
@@ -298,13 +316,21 @@ yjit_gen_block(ctx_t* ctx, block_t* block, rb_execution_context_t* ec)
 
     // For each instruction to compile
     for (;;) {
+        // Get the current pc and opcode
+        VALUE *pc = yjit_iseq_pc_at_idx(iseq, insn_idx);
+        int opcode = yjit_opcode_at_pc(iseq, pc);
+        RUBY_ASSERT(opcode >= 0 && opcode < VM_INSTRUCTION_SIZE);
+
+        // opt_getinlinecache wants to be in a block all on its own. Cut the block short
+        // if we run into it. See gen_opt_getinlinecache for details.
+        if (opcode == BIN(opt_getinlinecache) && insn_idx > starting_insn_idx) {
+            jit_jump_to_next_insn(&jit, ctx);
+            break;
+        }
```

Comment:
I don't like having stuff that's specific to individual instructions in this loop. Maybe we can just invalidate the whole block for now?

Note: eventually, we do need a single-pass analysis to end blocks at specific jump targets to fix an unrelated issue.

Reply:
> Maybe we can just invalidate the whole block for now?
How would you figure out the start of the block that contains the `opt_getinlinecache` instruction? It's possible to do a search for them among all blocks in the iseq, I guess...

Reply:
but why are you trying to make sure opt_getinlinecache is in its own block?

Reply:
It just has to be the first instruction in the block, that way the interpreter can find the block by looking at the version array in the iseq. It becomes its own block naturally because it returns END_BLOCK. We had a chat about this in our meeting, so I think you understand why now. Feel free to unresolve if it's still unclear.